<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro">
    <xacro:property name="M_PI" value="3.14159"/>
    <xacro:property name="M_SCALE" value="0.01"/>
    <xacro:property name="BASE_RADIUS" value="0.18"/>
    <xacro:property name="BASE_HEIGHT" value="0.05"/>
    <xacro:property name="BASE_MASS" value="20"/>

    <!-- robotino_body -->
    <xacro:macro name="robotino_body">
        
        <!-- base_link -->
        <link name="base_link">
            <inertial>
                <mass value="${BASE_MASS}" />
                <origin xyz="0 0 0" />
                <inertia ixx="1.0" ixy="0.0" ixz="0.0"
                    iyy="1.0" iyz="0.0"
                    izz="1.0" />
            </inertial>
            <visual>
                <origin xyz="0.01 0 0" rpy="0 0 ${M_PI/2}"/>
                <geometry>
                    <mesh filename="package://robotino_description/meshes/parts/robotino_base.dae"
                        scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
                </geometry>
            </visual>
            <collision>
                <origin xyz="0 0 ${BASE_HEIGHT/2}" rpy="0 0 0" />
                <geometry>
                    <cylinder radius="${BASE_RADIUS}" length="${BASE_HEIGHT}"/>
                </geometry>
            </collision>
        </link>
    </xacro:macro>
</robot>
